core/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -113,20 +116,22 @@
     /**
      * Those 3 state listeners are changing infrequently - CopyOnWriteArrayList is just fine
      */
-    private final Collection<ClusterStateListener> priorityClusterStateListeners = new CopyOnWriteArrayList<>();
-    private final Collection<ClusterStateListener> clusterStateListeners = new CopyOnWriteArrayList<>();
-    private final Collection<ClusterStateListener> lastClusterStateListeners = new CopyOnWriteArrayList<>();
+    private final Collection<ClusterStateApplier> priorityClusterStateAppliers = new CopyOnWriteArrayList<>();
```

Comment:
I would prefer to make the names of the variables consistent, i.e., `firstClusterStateAppliers` to make it the dual to `lastClusterStateAppliers`.
Another approach would be to name these:
- highPriorityStateAppliers
- normalPriorityStateAppliers
- lowPriorityStateAppliers

The `add` methods could then also be named:
- addHighPriorityStateApplier
- addStateApplier
- addLowPriorityStateApplier